scrapy/http/request/form.py

```diff
@@ -33,7 +33,14 @@ def __init__(self, *args, **kwargs):
                 self.headers.setdefault(b'Content-Type', b'application/x-www-form-urlencoded')
                 self._set_body(querystr)
             else:
-                self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
+                url_split = urlsplit(self.url)
+                formdata_key_list = []
+                for k in querystr.split('&'):
```

Comment:
It's better to use functions for URL parsing from standard library and w3lib instead of doing this manually here, unless they are not doing what you need here (in that case it would make sense to leave a comment)